integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java

```diff
@@ -59,67 +60,57 @@
     public Uni<FriesianCow> reactiveCowPersist() {
         final FriesianCow cow = new FriesianCow();
         cow.name = "Carolina";
-        return mutinySession
-                .persist(cow)
-                .chain(() -> mutinySession.flush())
-                .chain(s -> mutinySession.createQuery("from FriesianCow f where f.name = :name", FriesianCow.class)
-                        .setParameter("name", cow.name).getSingleResult());
+        return mutinySessionFactory.withTransaction((mutinySession, transaction) -> mutinySession.persist(cow))
+                .chain(() -> mutinySessionFactory.withTransaction((mutinySession, transaction) -> mutinySession
+                        .createQuery("from FriesianCow f where f.name = :name", FriesianCow.class)
+                        .setParameter("name", cow.name).getSingleResult()));
     }
 
     @GET
     @Path("/reactiveRemoveTransientEntity")
     public Uni<String> reactiveRemoveTransientEntity() {
         return populateDB()
                 .chain(() -> selectNameFromId(5))
-                .invoke(name -> {
-                    if (name == null)
+                .map(name -> {
+                    if (name == null) {
                         throw new AssertionError("Database was not populated properly");
+                    }
+                    return name;
                 })
-                .chain(() -> mutinySession.merge(new GuineaPig(5, "Aloi")))
-                .chain(aloi -> mutinySession.remove(aloi))
-                .chain(() -> mutinySession.flush())
+                .chain(() -> mutinySessionFactory
+                        .withTransaction((mutinySession, transaction) -> mutinySession.merge(new GuineaPig(5, "Aloi"))
+                                .chain(aloi -> mutinySession.remove(aloi))))
                 .chain(() -> selectNameFromId(5))
-                .map(result -> {
-                    if (result == null)
-                        return "OK";
-                    else
-                        return result;
-                });
+                .onItem().ifNotNull().transform(result -> result)
```

Comment:
I'm surprised this is necessary, since it's not doing anything?

Reply:
I forgot to look into that; there _was_ a reason, I don't remember what exactly - and it's existing code so let's look at that another time.